File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/B.Чётные_и_нечётные_числа/solution.cpp
#include <iostream>
#include <cmath>

int main() {
    /*
    Time Complexity O(1)
    Memory Complexity O(1)
    */
    int a, b, c;
    std::cin >> a >> b >> c;
    int diff = std::abs(a % 2) + std::abs(b % 2) + std::abs(c % 2);
    std::cout << (diff == 0 || diff == 3? "WIN" : "FAIL") << std::endl;
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/1_sprint/B.Чётные_и_нечётные_числа/solution.py
import sys


def solution(a: int, b: int, c: int) -> str:
    """
    Time Complexity O(1)
    Memory Complexity O(1)
    """
    diff: int = abs(a % 2) + abs(b % 2) + abs(c % 2)
    return 'WIN' if diff == 3 or diff == 0 else 'FAIL'


if __name__ == '__main__':
    a, b, c = map(int, sys.stdin.readline().rstrip().split())
    sys.stdout.write(solution(a, b, c))